src/fittable_laws.jl

```diff
@@ -80,3 +80,161 @@ function (law::FM90)(wave::T) where T <: Real
+    SIL1_amp::T = 0.002 * (1 / 3.08 + 1)
+    SIL1_lambda::T = 9.7
+    SIL1_b::T = -1.95
+    SIL1_n::T = 2.0
+    SIL2_amp::T = 0.002 * (1 / 3.08 + 1)
+    SIL2_lambda::T = 18.0
+    SIL2_b::T = -1.80
+    SIL2_n::T = 2.0
+    FIR_amp::T = 0.012 * (1 / 3.08 + 1)
+    FIR_lambda::T = 25.0
+    FIR_b::T = 0.00
+    FIR_n::T = 2.0
+    function P92(BKG_amp, BKG_lambda, BKG_b, BKG_n, FUV_amp, FUV_lambda, FUV_b, FUV_n,
+                 NUV_amp, NUV_lambda, NUV_b, NUV_n, SIL1_amp, SIL1_lambda, SIL1_b, SIL1_n,
+                 SIL2_amp, SIL2_lambda, SIL2_b, SIL2_n, FIR_amp, FIR_lambda, FIR_b, FIR_n)
+
+        BKG_amp ≥ 0 || error("`BKG_amp` must be ≥ 0, got ", BKG_amp)
+        FUV_amp ≥ 0 || error("`FUV_amp` must be ≥ 0, got ", FUV_amp)
+        0.06 ≤ FUV_lambda ≤ 0.08 || error("`FUV_lambda` must be in between [0.06, 0.08], got ", FUV_lambda)
+        NUV_amp ≥ 0 || error("`NUV_amp` must be ≥ 0, got", NUV_amp)
+        0.20 ≤ NUV_lambda ≤ 0.24 || error("`NUV_lambda` must be in between [0.20, 0.24], got ", NUV_lambda)
+        SIL1_amp ≥ 0 || error("`SIL1_amp` must be ≥ 0, got ", SIL1_amp)
+        7 ≤ SIL1_lambda ≤ 13 || error("`SIL1_lambda` must be in between [7, 13], got ", SIL1_lambda)
+        SIL2_amp ≥ 0 || error("`SIL2_amp` must be ≥ 0, got ", SIL2_amp)
+        15 ≤ SIL2_lambda ≤ 21 || error("`SIL2_lambda` must be in between [15, 21], got ", SIL2_lambda)
+        FIR_amp ≥ 0 || error("`FIR_amp` must be ≥ 0, got ", FIR_amp)
+        20 ≤ FIR_lambda ≤ 30 || error("`FIR_lambda` must be in between [20, 30], got ", FIR_lambda)
+
+        params = promote(BKG_amp, BKG_lambda, BKG_b, BKG_n,
+                         FUV_amp, FUV_lambda, FUV_b, FUV_n,
+                         NUV_amp, NUV_lambda, NUV_b, NUV_n,
+                         SIL1_amp, SIL1_lambda, SIL1_b, SIL1_n,
+                         SIL2_amp, SIL2_lambda, SIL2_b, SIL2_n,
+                         FIR_amp, FIR_lambda, FIR_b, FIR_n)
+
+        return new{eltype(params)}(params...)
+    end
+end
+
+bounds(::Type{<:P92}) = (10, 10_000_000)
+
+function (law::P92)(wave::T) where T <: Real
+    checkbounds(law, wave) || return zero(float(T))
+
+    x = aa_to_invum(wave)
+    lam = 1.0 / x # wavelength is in microns
+
+    axav = _p92_single_term(lam, law.BKG_amp, law.BKG_lambda, law.BKG_b, law.BKG_n)
+    axav += _p92_single_term(lam, law.FUV_amp, law.FUV_lambda, law.FUV_b, law.FUV_n)
+    axav += _p92_single_term(lam, law.NUV_amp, law.NUV_lambda, law.NUV_b, law.NUV_n)
+    axav += _p92_single_term(lam, law.SIL1_amp, law.SIL1_lambda, law.SIL1_b, law.SIL1_n)
+    axav += _p92_single_term(lam, law.SIL2_amp, law.SIL2_lambda, law.SIL2_b, law.SIL2_n)
+    axav += _p92_single_term(lam, law.FIR_amp, law.FIR_lambda, law.FIR_b, law.FIR_n)
+
+    return axav
+end
+
+# function for calculating a single P92 term
+function _p92_single_term(x::Real, amplitude::Real, cen_wave::Real, b::Real, n::Real)
+    l_norm = x / cen_wave
+    return amplitude / (l_norm^n + inv(l_norm^n) + b)
+end
```

Comment:
```suggestion
function _p92_single_term(x::Real, amplitude::Real, cen_wave::Real, b::Real, n::Real)
    l_norm = x / cen_wave
    l_norm_n = l_norm^n
    return amplitude / (l_norm_n + inv(l_norm_n) + b)
end
```
Computing `l_norm^n` once and reusing it improves performance